# Single-cell molecular QTL mapping
  
Following data QC and preparation, we conduct single-cell molecular QTL mapping. For now, this is focused on eQTL mapping.  
For an overview of the whole process, please see the (MMC powerpoint)[https://uoe-my.sharepoint.com/:p:/r/personal/kcampb2_ed_ac_uk/Documents/MMC_PPA_notes.pptx?d=w3afaa31d5ccd4efea7dff2bf2b120465&csf=1&web=1&e=jjdB1P]
  
## eQTL mapping  
  
For an easy overview of the proposed analysis, please see [eQTL_onepager.md](eQTL_onepager.md). This document includes the **project goal/overview**, **functional and design specifications** of the analysis pipeline and the **analysis plan and timeline**.   
  
## Future analysis ideas  
  
Please feel free to add any and all ideas you would like to implement (or see implemented by the team/me)!  
  
- Variance QTL mapping  
- Dynamic cell state/continuous cell state or phenotype mapping (perhaps go straight into this?)  
- Allelic imbalance analysis (see [reading list](reading_list.md))  
- Something with reads or tails of reads?   
- Producing a sc-eQTL atlas (similar to cattle gtex)  
- Which genes have the most variation across a single cell type? (and which SNPs/regions is this associated with) --> varQTLs