# File Structure After Analysis

## Overview

This guide describes the expected file structure after running the arcas pipeline, performing QC and filtering, and carrying out Seurat processing to create finalised Seurat objects ready for downstream analysis.

---

## File Structure

analysis/					# Base directory for running the analysis
├config/					
│  └config.yaml

├data/
│  ├reference/
│  │  └homo_sapiens/GRCH38/
│  │      ├reference.dna.fa.gz
│  │      └reference.cdna.fa.gz
│  ├reads/					# Input .fastq.gz files go here
│  ├samples.tsv					# Input - See USAGE.md for requirements
│  └run_sheet.tsv				# Generated by 00.1 script

├scripts/
│  ├00_preprocessing/
│  │  ├00.1_make_run_sheet.sh			# Auto-generates run_sheet.tsv from fastqs
│  │  └00.2_dag.sh				# Generates DAG of arcas pipeline
│  ├01_arcas_pipeline/
│  │  └01.1_run_pipeline.slurm			# Runs entire arcas pipeline - LONG
│  ├02_QC_and_filtering/
│  │  ├02.1_QC_and_filtering_pipeline.slurm	# Runs all QC and filtering scripts as a pipeline
│  │  ├02.1.1_Snakefile_bustools		# Converts kallisto.sorted_bus to sorted_bus.txt for each sample
│  │  ├02.1.2_mapping_stats_command.R		# Mapping stats for samples and flowcells
│  │  ├02.1.3_busfile_QC_command.R		# QC metrics for sorted_bus.txt files
│  │  ├02.1.4_seurat_QC_command.R		# Initial Seurat QC and filtering
│  │  └02.1.5_cells_and_reads_command.R		# Number of cells and reads QC
│  └03_seurat_processing/
│     ├03.1_seurat_processing_pipeline.slurm	# Runs all seurat scripts as pipeline
│     ├03.1.1_seurat_process_command.R		# Split to tissue specific, integrate
│     └03.1.2_seurat_finalise_command.R		# Process tissue specific to create analysis ready seurat objects

├outputs/
│  ├arcas_pipeline/
│  │  └<runname>/MMC/
│  │      ├<sample>/
│  │      │  ├kallisto.*			# kallisto outputs inc busfiles, ec, json with run info
│  │      │  ├count.genes.*.mtx			# Count matricies for mature, nascent and ambiguous genes
│  │      │  ├count.genes.names.txt		# Transcript IDs for count matricies
│  │      │  ├QC/
│  │	  │  │  ├UMI_per_barcode.png
│  │	  │  │  ├knee_plot.png
│  │	  │  │  └barcode_summary.tsv
│  │	  │  ├Seurat/
│  │	  │  │  ├unprocessed_seurat.rds		# Sample specific pre-filtering seurat object
│  │	  │  │  ├seurat_prefilter_vln.png
│  │	  │  │  └seurat_prefilter_count_v_gene.png
│  │	  ├parse.wells.demuplx/			# Split fastqs per sample/well
│  │	  │  ├<well>.(bu|cdna1|cdna2).fastq.gz
│  │	  │  └<well>_(0|1|2).fastq.gz
│  │	  ├runs/
│  │	  │  ├<fastq>.(bu|cdna1|cdna2).fastq.gz # Input fastqs split into barcode/umi, and non technical parts of read 1 and read 2
│  │      │  ├<fastq>.fastp.(html|json)		# fastp outputs
│  │      │  ├<fastq>.r(1|2)_fastqc.(html|zip)  # fastQC outputs
│  │      │  ├<fastq>.qc.r(1|2).fastq.gz	# QC for each read
│  │      │  ├<fastq>.unassigned.r(1|2).fastq.gz# Unbarcoded reads
│  │      │  ├<fastq>.extract.summary.json	# Info on n_reads, n_barcoded
│  │      │  └parse.WT_mega.v3.read.split.config# Demultiplexing config file
│  │	  ├bu.fastq.gz				# Barcodes / UMI for all
│  │	  ├cdna1.fastq.gz			# Non-technical parts of all R1
│  │	  ├cdna2.fastq.gz			# Non-technical parts of all R2
│  │	  ├parse.wells.demuplx.config
│  │	  └parse.wells.demuplx.summary.json
│  ├QC_and_filtering/
│  │  ├cells_per_sample.png
│  │  ├cells_per_sample_fc.png			# Split by flowcell
│  │  ├cells_per_sample_tissue.png		# Split by tissue
│  │  ├library_complexity.png			# Proportion of unique molecules
│  │  ├reads_per_cell.png
│  │  ├reads_per_cell_fc.png
│  │  ├reads_per_cell_tissue.png
│  │  ├UMIs_per_barcode_per_sample.png
│  │  ├knee_per_sample
│  │  ├knee_filter_outcome_per_sample.png	# Cells kept/removed if knee filter used
│  │  ├busfile_qc_metrics.csv
│  │  └mapping_stats/
│  │	  ├flowcell_mapping_stats.(png|csv)
│  │	  └sample_mapping_stats.(png|csv)
│  └seurat_processing/
│     ├intermediate/				# Seurat objects saved at each stage of processing prior to final objects
│     │	  ├seurat_list.rds
│     │	  ├seurat_list_filtered.rds
│     │	  ├seurat_list_filtered_normalised_variable.rds
│     │	  ├<tissue>_integrated.rds
│     │	  └<tissue>_anchors.rds
│     ├metadata/				# Metadata and markers extracted from final seurat objects
│     │   ├<tissue>_metadata.csv
│     │   └<tissue>_all_markers.rds
│     ├figures/
│     │   ├qc/					# Figures for all tissues (prior to split)
│     │   │  ├seurat_violin_qc.png
│     │   │  ├seurat_filtering.png		# Cells kept / removed by filtering steps
│     │   │  ├seurat_nCount_RNA.png
│     │   │  ├seurat_nFeature_RNA.png
│     │   │  └seurat_percent.mt.png
│     │   └<tissue>/
│     │      ├<tissue>_umap_clusters.png
│     │      ├<tissue>_umap_condition.png
│     │      ├<tissue>_umap_sample.png
│     │      ├<tissue>_umap_flowcell.png
│     │      ├<tissue>_pca_clusters.png
│     │      └<tissue>_violinplot_qc.png
│     └analysis_ready/				# Final cleaned and integrated objects ready for downstream analysis
│            └<tissue>_integrated_final.rds
